### DISCUSSION TITLE
Training and inference with T2-only files

### DISCUSSION THREAD

**Discussion by paul-reiners** [MIC-DKFZ/nnUNet#1186] [Q&A]
Do you know how when you run the T2-only inference model, it expects the file to end with “0000.nii.gz” even though it’s a T2? How heavy of a lift would it be to change this so that it expects the file to end with “0001.nii.gz?”

I have the following in my dataset.json file:

```
    "modality": {
        "1": "T2"
    },

```
The T2 files all end in ".0001.nii.gz". There are no T1 files.  Yet, I get this error when running nnUNet_plan_and_preprocess.py:

```
Traceback (most recent call last):
  File "/home/miran045/reine097/projects/nnUNet/nnunet/experiment_planning/nnUNet_plan_and_preprocess.py", line 170, in <module>
    main()
  File "/home/miran045/reine097/projects/nnUNet/nnunet/experiment_planning/nnUNet_plan_and_preprocess.py", line 105, in main
    verify_dataset_integrity(join(nnUNet_raw_data, task_name))
  File "/home/miran045/reine097/projects/nnUNet/nnunet/preprocessing/sanity_checks.py", line 133, in verify_dataset_integrity
    c, expected_image_files)
AssertionError: some image files are missing for case 0mo_template_01. Expected files:
 ['/home/feczk001/shared/data/nnUNet/nnUNet_raw_data_base/nnUNet_raw_data/Task540_T2_only_Fold0/imagesTr/0mo_template_01_0000.nii.gz']
Verifying training set
checking case 0mo_template_01

```

**Comment by Joeycho** [comment]
Yes, I guess you should name it with .0000.nii.gz, since there is only one modality. (0, 1, 2, 3, 4... sequentially).

It doesn't matter whether it is PET, MR T1, MR T2, Ultrasound, CT... if there is only one modality then, it is with .0000.nii.gz. Just be aware of mentioning in dataset.json file, 0 means T2, 1 means.. other... etc.

But I understand... this part is quite strict in nnUNet framework.. and it is one of my motivation to try to move on with my own dataloader.. data preprocessing..

**Comment by Alicegif** [comment]
The version I am using in 2026 doesn't seem to care. I originally thought I would be combining multiple modalities in one model so I put CT as 0004 as my only label and it worked on multiple datasets. No placeholders with zeroed out files for 0,1,2,3.  I screwed up on another dataset and named them ending with 1, 2, 3 instead of 0, 1, 2 and it is training.
